code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -0,0 +1,213 @@
+
+/obj/effect/landmark/corpse/mob_spawn/seed_vault/special(mob/living/new_spawn)
+	var/plant_name = pick("Tomato", "Potato", "Brocolli", "Carrot", "Deathcap", "Ambrosia", "Pumpkin", "Ivy", "Kudzu", "Bannana", "Moss", "Flower", "Bloom", "Spore", "Root", "Bark", "Glowshroom", "Petal", "Leaf", "Venus", "Sprout","Cocao", "Strawberry", "Citrus", "Oak", "Cactus", "Pepper")
+	new_spawn.real_name = plant_name
+
+//Greed
+
+/obj/structure/cursed_slot_machine
+	name = "greed's slot machine"
+	desc = "High stakes, high rewards."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "slots-on"
+	anchored = 1
+	density = 1
+	var/win_prob = 5
+/*
+/obj/structure/cursed_slot_machine/attack_hand(mob/living/carbon/human/user)
+	if(!istype(user))
+		return
+	if(in_use)
+		return
+	in_use = TRUE
+	user << "<span class='danger'><B>You feel your very life draining away as you pull the lever...it'll be worth it though, right?</B></span>"
+	user.adjustCloneLoss(20)
+	if(user.stat)
+		user.gib()
+	icon_state = "slots2"
+	sleep(50)
+	icon_state = "slots1"
+	in_use = FALSE
+	if(prob(win_prob))
+		new /obj/item/weapon/dice/d20/fate/one_use(get_turf(src))
+		if(user)
+			user << "You hear laughter echoing around you as the machine fades away. In it's place...more gambling."
+			qdel(src)
+	else
+		if(user)
+			user << "<span class='danger'>Looks like you didn't win anything this time...next time though, right?</span>"
+*/
+//Gluttony
+
+/obj/effect/gluttony
+	name = "gluttony's wall"
+	desc = "Only those who truly indulge may pass."
+	anchored = 1
+	density = 1
+	icon_state = "blob"
+	icon = 'icons/mob/blob.dmi'
+
+/obj/effect/gluttony/CanPass(atom/movable/mover, turf/target, height=0)//So bullets will fly over and stuff.
+	if(height==0)
+		return 1
+	if(istype(mover, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = mover
+		if(H.nutrition >= NUTRITION_LEVEL_FAT)
+			return 1
+		else
+			H << "<span class='danger'><B>You're not gluttonous enough to pass this barrier!</B></span>"
+	else
+		return 0
+
+//Pride
+
+/obj/structure/mirror/magic/pride
+	name = "pride's mirror"
+	desc = "Pride cometh before the..."
+	icon_state = "magic_mirror"
+/* TODO:curse and chasm
+/obj/structure/mirror/magic/pride/curse(mob/user)
+	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as their hand leaves the mirror!</B></span>")
+	var/turf/T = get_turf(user)
+	T.ChangeTurf(/turf/simulated/chasm/straight_down)
+	var/turf/simulated/chasm/straight_down/C = T
+	C.drop(user)*/
+
+//Sloth - I'll finish this item later
+
+//Envy
+
+/obj/item/weapon/knife/envy
+	name = "envy's knife"
+	desc = "Their success will be yours."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "render"
+	item_state = "render"
+	force = 18
+	throwforce = 10
+	w_class = 3
```

Comment:
please use the weight class defines: `WEIGHT_CLASS_NORMAL` in this instance I do believe.